From 10f1427fa2c84e6efc348091d77c24f02c6cff30 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Mon, 19 Oct 2026 21:32:56 -0700
Subject: [PATCH] fixup! rpc(v2): adapt blockchain, mempool and mining
 endpoints

Check the snapshot template native payout through its committed script, rather than constructing an implicit legacy-to-native output conversion. Keep header, transaction and state checks. Focused RPC snapshot mining8cases pass.

Tier: V2
---
 src/test/rpc_snapshot_mining_tests.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/test/rpc_snapshot_mining_tests.cpp b/src/test/rpc_snapshot_mining_tests.cpp
index ec1ce81444..6cae9d8636 100644
--- a/src/test/rpc_snapshot_mining_tests.cpp
+++ b/src/test/rpc_snapshot_mining_tests.cpp
@@ -2,6 +2,7 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
+#include <primitives/v2_outputs.h>
 #include <config.h>
 #include <consensus/amount.h>
 #include <consensus/merkle.h>
@@ -335,8 +336,7 @@ BOOST_AUTO_TEST_CASE(getblocktemplate_mineraddress_and_submitsealedblock) {
     CMutableTransaction decoded_cb;
     BOOST_REQUIRE(DecodeHexTx(decoded_cb, cb_hex));
     BOOST_REQUIRE(!decoded_cb.vout.empty());
-    BOOST_CHECK(decoded_cb.vout[0].ToV2Output() ==
-        CTxOut(decoded_cb.vout[0].nValue, expectedMinerSpk).ToV2Output());
+    BOOST_CHECK(CoinbaseOutputPaysScript(decoded_cb.vout[0], expectedMinerSpk));
 
     CBlockHeader header;
     BOOST_REQUIRE(DecodeHexBlockHeader(header, header_hex));
-- 
2.50.1 (Apple Git-155)

